app/controllers/api/v1/trainers/pokemons_controller.rb

```diff
@@ -4,33 +4,39 @@ def create
     trainer_id = params[:trainer_id]
     begin
       pokemon = Pokemon.populate_with_api_data(pokemon_params, pokemon_name, trainer_id)
-      if pokemon.save
-        render json: PokemonSerializer.new(pokemon), status: :created
-      else
-        render json: ErrorSerializer.format_error(pokemon.errors, 422), status: :unprocessable_entity
-      end
+      render json: PokemonSerializer.new(pokemon), status: :created if pokemon.save
     rescue StandardError => e
-      render json: ErrorSerializer.format_error(e, 422), status: :unprocessable_entity
+      render json: ErrorSerializer.format_general_error(e, 422), status: :unprocessable_entity
```

Comment:
Very helpful to include a general and specific error handling methods in our serializer, definitely clarifies errors for users and those reviewing.